src/ghc.jl

```diff
@@ -0,0 +1,133 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    GHC(kernel,merge_crit;k=35,vars=nothing)
+
+A method for partitioning spatial data into `k` clusters 
+using Geostatistical Hierarchical Clustering (GHC). This method generates
+clusters by assigning a unique cluster to each datum, 
+iteratively joining the two clusters which are most similar
+according to a kernel estimator on the data's spatial 
+dependence, and finally stopping this process at the step 
+where the clusters' silhouette index is optimized.
+
+## Parameters
+
+* `kernel`      - type of kernel used for estimator
+* `merge_crit`  - strategy to specify difference between clusters
+* `k`           - minimum number of observations contained in support 
+                    of kernel function centered at each point. 
+                    Defaults to 35.
+* `vars`        - Variables (or features) to consider (default to all)
+
+## References
+* F. Fouedjio. 2016. [A hierarchical clustering method for multivariate geostatistical data]
+  (https://www.sciencedirect.com/science/article/abs/pii/S2211675316300367)
+"""
+struct GHC <: PartitionMethod
+  kernel::AbstractString
+  merge_crit::AbstractString
+  k::Int
+  λ::Float64
+  vars::Union{Vector{Symbol},Nothing}
+end
+
+GHC(kernel::AbstractString, merge_crit::AbstractString; k=35,λ=10, vars=nothing) =
+  GHC(kernel, merge_crit, k,λ, vars)
```

Comment:
We can omit the type annotations. It is usually a good idea because it makes the code cleaner. Type annotations are only for cases when you want to disambiguate two different constructors for example for the compiler.